src/components/Syntaxes.jsx

```diff
@@ -24,15 +33,35 @@ class Syntaxes extends React.Component {
     this.props.onClick(syntax);
   }
 
-  render() {
+  getRelevantSyntaxes(searchSyntax) {
     const { syntaxes } = this.props;
+    const escapedSearchSyntax = regExpEscape(searchSyntax);
+    const regExp = new RegExp(escapedSearchSyntax, 'gi');
+
+    return escapedSearchSyntax ? syntaxes.filter(syntax => syntax.match(regExp)) : syntaxes;
+  }
+
+  filterSyntaxes(event) {
+    const searchSyntax = event.target.value.trim();
```

Comment:
I would rather separate input state from our syntaxes states. So on input change we will set only its value and inside render we can filter all syntaxes from props by our input value